src/map/clif.c

```diff
@@ -4566,7 +4566,8 @@ static void clif_tradeadditem(struct map_session_data *sd, struct map_session_da
 		else
 			p.itemId = sd->status.inventory[index].nameid;
 #if PACKETVER >= 20100223
-		p.itemType = sd->inventory_data[index]->type;
+		if (sd->inventory_data[index] != NULL)
```

Comment:
this is already being checked few lines above, you can probably replace both checks with a nullpo_retv under the assert.